components/SensorComponent.js

```diff
@@ -75,22 +96,54 @@ export const SensorComponent = () => {
     return () => _unsubscribe();
   }, []);
 
+  const [animal, setAnimal] = useState({});
+  const [loading, setLoading] = useState(false);
+
+  const generateAnimal = () => {
+    setLoading(true);
+    fetch('https://zoo-animal-api.herokuapp.com/animals/rand')
+      .then((res) => res.json())
+      .then((data) => setAnimal(data))
+      .finally(() => setLoading(false));
+  };
+
+  useEffect(() => {
+    if (isShaking(data)) {
+      generateAnimal();
+    }
+  }, [data]);
+
+  const toShare = async () => {
+    try {
+      const result = await Share.share({
+        message: 'Hey, come take a look at a random animal!',
```

Comment:
Maybe here you could have added the picture {animal.image_link} or the link to the app so we could see the animal. Not so sure on how, tried the easiest way above it didn't work. 🙈

But bonus points for doing share and shake! 🙌